internal/api/v20240610preview/hcpopenshiftclusters_methods.go

```diff
@@ -0,0 +1,599 @@
+package v20240610preview
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"net/http"
+
+	configv1 "github.com/openshift/api/config/v1"
+
+	"github.com/Azure/ARO-HCP/internal/api"
+	"github.com/Azure/ARO-HCP/internal/api/arm"
+	"github.com/Azure/ARO-HCP/internal/api/v20240610preview/generated"
+)
+
+type HcpOpenShiftClusterResource struct {
+	generated.HcpOpenShiftClusterResource
+}
+
+type VersionProfile struct {
+	generated.VersionProfile
+}
```

Comment:
We don't strictly need these wrapper types.  I was being a little too cute with all the `Normalize` methods I slapped on everything.  Like in this case we could get by with just the `normalizeVersion` function and drop the method.  But we can defer that to a follow-up pull request.

Reply:
+1 for a follow up, this PR is already large